### PR TITLE
call init on new controls

### PR DESCRIPTION
**Pull request by chenz4027**
Signed-off-by: John Swanke <jswanke@redhat.com>
Signed-off-by: Maggie Chen <magchen@redhat.com>

https://issues.redhat.com/browse/ACM-1640

### CODE REVIEW

**Review by fxiang1 — DISMISSED** [review]
LGTM! Thanks Maggie!

**Comment by fxiang1** [comment]
I don't have admin access. We'll probably need @KevinFCormier to do the override.

**Comment by KevinFCormier** [comment]
@chenz4027 Would it be possible to cover these lines?
image

The first line is partially covered because of all the optional chaining, but it looks like the next lines are never executed in the tests.

I think this is slightly different from Jorge's example because these lines seem central to your fix and they are introducing new logic. He described a case where he was demotivated from doing a bit of refactoring or code clean-up in his PR because he would need to cover those areas as well.

If there is an impediment to covering these, or if you cover them and we are still not at threshold, let me know and I will merge. Thanks.

File: frontend/src/routes/Infrastructure/Clusters/ClusterPools/CreateClusterPool/CreateClusterPool.tsx [hunk 1]
```diff
@@ -230,16 +230,6 @@ export function CreateClusterPool() {
                     control.validation.required = mustJoinClusterSet ?? false
                 }
                 break
-            case 'infrastructure':
```
Comment (fxiang1) [inline frontend/src/routes/Infrastructure/Clusters/ClusterPools/CreateClusterPool/CreateClusterPool.tsx:233]: @chenz4027  I think we still need this though. This is not related to the reviewinfo.

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx [hunk 2]
```diff
@@ -299,16 +299,6 @@ export default function CreateClusterPage() {
                     control.validation.required = mustJoinClusterSet ?? false
                 }
                 break
-            case 'infrastructure':
```
Comment (fxiang1) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx:302]: Same for this code. Still need this.

**Review by fxiang1 — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ClusterPools/CreateClusterPool/CreateClusterPool.tsx (hunk 1 above)
Comment (KevinFCormier) [reply frontend/src/routes/Infrastructure/Clusters/ClusterPools/CreateClusterPool/CreateClusterPool.tsx:233]: @fxiang1 I think this is vestigial code from when the infrastructure was selected as the first step of the wizard. Can you demonstrate when this code would run and have an effect?

Comment (fxiang1) [reply frontend/src/routes/Infrastructure/Clusters/ClusterPools/CreateClusterPool/CreateClusterPool.tsx:233]: Yep you're right, we can go ahead and remove it. Thanks!

**Review by fxiang1 — APPROVED** [review]
LGTM! Thanks Maggie!

**Comment by chenz4027** [comment]
Not sure why the mirror image job failed. Will retest all

/retest
